src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  removeAnimal(animal) {
+    const animalIndex = Animal.alive.indexOf(animal);
+
+    Animal.alive.splice(animalIndex, 1);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Carnivore || herbivore.hidden) {
+      return;
+    }
+
+    herbivore.health -= 50;
+
+    if (herbivore.health <= 0) {
+      this.removeAnimal(herbivore);
```

Comment:
Instead of calling `this.removeAnimal`, you can directly filter `Animal.alive` inside the `bite` method after decreasing the health of the herbivore. This will remove the need for the `removeAnimal` method and simplify the logic.